tests/test-core.js

```diff
@@ -134,4 +134,12 @@ w.write(inline_content.slice(10), null)
 let actual_checksum = w.finish(null)
 assertEquals(actual_checksum, networks_checksum)
 
+// Basic locking API sanity test
+repo.lock_push(OSTree.RepoLockType.SHARED, null);
+repo.lock_push(OSTree.RepoLockType.SHARED, null);
+repo.lock_pop(null);
+repo.lock_pop(null);
+repo.lock_push(OSTree.RepoLockType.EXCLUSIVE, null);
+repo.lock_pop(null);
```

Comment:
Since this wasn't public, I ended up writing basically the same locking implementation in python for our (sadly internal) tooling. I ended up writing a bunch of tests for it and I'd say a major improvement is to open another `OstreeRepo` so you have 2 different threads competing for the lock (rather than trying to write a multi-threaded test from a single `OstreeRepo`. So:
```
let repo2 = OSTree.Repo.new(repoPath);
repo2.open(null);
repo.lock_push(...)
repo2.lock_push(...)
```

Another thing I'd strongly suggest is setting the repo lock timeout to `0` so that any issues here don't just block the test suite for 30 seconds:
```
config = repo.copy_config()
config.set_integer("core", "lock-timeout-secs", 0)
repo.write_config(config)
```
I'd like to be able to change the locking configuration at runtime without using the repo configuration, but that can be handled later.

Finally, testing the actual error states would be great. I think this is how it would go in JS:
```
repo.lock_push(OSTree.RepoLockType.EXCLUSIVE, null);
assertThrows("Locking repo", {
  repo2.lock_push(OSTree.RepoLockType.EXCLUSIVE, null);
})
```